.github/pull_request_template.md

```diff
@@ -8,6 +8,7 @@
 - [ ] This is a refactoring of components with existing test coverage.
 - [ ] Instructions for manual testing are as follows:
   1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]
+- [ ] New or changed functionality is sufficiently documented
```

Comment:
I think this should go in a separate checklist. See also https://github.com/galaxyproject/galaxy/pull/14871 and my comment there.

Reply:
Like so?

```suggestion

## Checklist

- [ ] New or changed functionality is sufficiently documented
```

Reply:
Documentation is very important and good documentation saves hours, days (weeks?) of time wasted on "code archaeology" and just plain frustration. But I'm not sure a PR template is the right place for encouraging contributors to write documentation. Here's why:
1. Lots of times that's simply not applicable for the type of PR. But if there's a documentation box I don't check, I feel somehow guilty, no?
2. Sometimes, bundling a tangential documentation improvement into an otherwise focused and concise modification will make the PR less atomic. Even more importantly, reviewing complex PRs is easier when they do not include additional improvements not related to the core edits.
3. Often we will want to first submit the PR, have the code reviewed, make changes, and only once it's been merged into dev are we ready to write any documentation. Writing good documentation takes time and effort; doing that for code that hasn't passed a review is premature, I think. 
4. (Good) documentation comes in different shapes and sizes. There's self-documenting code that's a pleasure to read; there's appropriate inline commenting, there are docstrings at different levels, there are READMEs (e.g. client, migrations), there's in-depth description of architectural and/or implementation choices in the PR description, there's formal online documentation, and then there's Galaxy architecture slides and a variety of tutorials. I doubt a checkbox is an adequate representation of all that, or an effective way to encourage contributors to "do their part."

Overall, +100 on an urgent need for much better documentation at all levels. But I doubt a PR template is the best way to address it.